### PR TITLE
Document ADR for denying cross-ns refs by default

### PR DESCRIPTION
**Pull request by squaremo**
This PR introduces a decision record for changing the default treatment of cross-namespace references, which appear in a couple of places in the API.

This is a reversal of the (implicit) decision to respect cross-namespace references by default, so deserves to be recorded.

### CODE REVIEW

**Comment by squaremo** [comment]
@chanwit @Nalum @phoban01 @tomhuang12 As maintainers, would you check you're happy with this decision. With reference to https://github.com/weaveworks/tf-controller/blob/main/rfcs/DECISION_MAKING.md, I think lazy consensus is enough here; I'll merge in a couple of days, assuming at least one person approves the PR.

**Review by phoban01 — APPROVED** [review]

**Review by Nalum — APPROVED** [review]

**Review by chanwit — APPROVED** [review]
Thank you @squaremo 
LGTM

**Pull request merged**
